From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Dan Muey <daniel.muey@webpros.com>
Date: Thu, 22 May 2025 14:59:41 +0000
Subject: [PATCH] PCRE config RPATH adjustment

Case EA-6376: mod_security2.so has faulty RPATH
Updated find_pcre.m4 to handle libpcre.so installed in system
library search paths.
---
 build/find_pcre.m4 | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/build/find_pcre.m4 b/build/find_pcre.m4
index 9b222f6..8676483 100644
--- a/build/find_pcre.m4
+++ b/build/find_pcre.m4
@@ -68,6 +68,9 @@ else
         PCRE_LDADD="`${PCRE_CONFIG} --libs`"
         if test "$verbose_output" -eq 1; then AC_MSG_NOTICE(pcre LDADD: $PCRE_LDADD); fi
         PCRE_LD_PATH="/`${PCRE_CONFIG} --libs | cut -d'/' -f2,3,4,5,6 | cut -d ' ' -f1`"
+        if test "x`${PCRE_CONFIG} --libs | grep /`" != "x"; then
+            PCRE_LD_PATH="-R /`${PCRE_CONFIG} --libs | cut -d'/' -f2,3,4,5,6 | cut -d ' ' -f1`"
+        fi
         if test "$verbose_output" -eq 1; then AC_MSG_NOTICE(pcre PCRE_LD_PATH: $PCRE_LD_PATH); fi
     else
         AC_MSG_RESULT([no])
